src/index.js

```diff
@@ -15,15 +15,16 @@ export function sampleRUM(checkpoint, data) {
   const timeShift = () => (window.performance ? window.performance.now() : Date.now() - window.hlx.rum.firstReadTime);
   try {
     window.hlx = window.hlx || {};
-    if (!window.hlx.rum) {
+    if (!window.hlx.rum || (window.hlx.rum && typeof window.hlx.rum.collector === 'undefined')) {
```

Comment:
why `typeof` is needed here ? 
no need to re-test for `window.hlx.rum` after the OR... 

```suggestion
    if (!window.hlx.rum || typeof window.hlx.rum.collector === 'undefined') {
```

or even maybe 

```suggestion
    if (!window.hlx.rum || !window.hlx.rum.collector) {
```